eng/common/scripts/SemVer.ps1

```diff
@@ -132,8 +132,12 @@ class AzureEngSemanticVersion {
 
     if ($this.IsPrerelease)
     {
-      $versionString += $this.PrereleaseLabelSeparator + $this.PrereleaseLabel + `
-                        $this.PrereleaseNumberSeparator + $this.PrereleaseNumber
+      $versionString += $this.PrereleaseLabelSeparator + $this.PrereleaseLabel
```

Comment:
@chidozieononiwu why are we removing change log entries when the version doesn't match? Shouldn't we keep them even if we don't match?

@praveenkuttappan how does fixing this ToString fix the removal of the entries in the changelog? While I don't mind adding this support if there is a good reason I'd prefer not supporting scenarios like this that don't match our guidelines if we can avoid it.

Reply:
Version string for these old prerelease versions is updated with prerelease number 0  when versions are sorted and this causes a mismatch. Update change log code looks for change log for each of the sorted versions and not finding the one in older format due to this mismatch. for e.g. Key in map is 10.2.0-preview and change log update code is looking for 10.2.0-preview0. Fix is resolving this issue by not adding prerelease number if it is 0 so it can support older versions too.

Reply:
Thanks @praveenkuttappan . This should resolve this as far as converting ToString() and sorting the version string. But the version itself is still parsed wrongly. Also maybe it a bad idea for Update-Changelog to depend on the semvar sorting due to older no compliant version. But as long as ToString() returns the original value parsed then it should fix it in all cases, Although I wonder if it might be the case that the sorting is now affected.

Reply:
I'd like to solve this issue differently. Instead of loosening our version parsing here we should just have the changelog parsing code to not sort these entries with invalid format and just leave them unchanged.

Reply:
Any idea how I can go about this. We parse out the changelog into Hashtables. Am using the sorting to make sure I write it back in the correct order. I guess I could use an ordered HashTable for the whole process then we wouldn't have to worry about sorting.